go.mod

```diff
@@ -10,33 +10,33 @@ require (
 	github.com/mochi-mqtt/server/v2 v2.6.5
 	github.com/onsi/ginkgo/v2 v2.20.0
 	github.com/onsi/gomega v1.34.1
-	github.com/openshift/api v0.0.0-20240527133614-ba11c1587003
-	github.com/openshift/build-machinery-go v0.0.0-20240419090851-af9c868bcf52
-	github.com/openshift/library-go v0.0.0-20240621150525-4bb4238aef81
+	github.com/openshift/api v0.0.0-20241001152557-e415140e5d5f
+	github.com/openshift/build-machinery-go v0.0.0-20240613134303-8359781da660
+	github.com/openshift/library-go v0.0.0-20241107160307-0064ad7bd060
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/client_golang v1.18.0
+	github.com/prometheus/client_golang v1.19.1
 	github.com/spf13/cobra v1.8.1
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.9.0
 	github.com/valyala/fasttemplate v1.2.2
 	golang.org/x/exp v0.0.0-20240719175910-8a7402abbf56
-	golang.org/x/net v0.28.0
+	golang.org/x/net v0.29.0
 	gopkg.in/yaml.v2 v2.4.0
 	helm.sh/helm/v3 v3.15.3
-	k8s.io/api v0.30.3
-	k8s.io/apiextensions-apiserver v0.30.3
-	k8s.io/apimachinery v0.30.3
-	k8s.io/apiserver v0.30.3
-	k8s.io/client-go v0.30.3
-	k8s.io/component-base v0.30.3
+	k8s.io/api v0.31.1
+	k8s.io/apiextensions-apiserver v0.31.1
+	k8s.io/apimachinery v0.31.2
```

Comment:
why this is 0.31.2 and others are 0.31.1 :)

Reply:
0.31.1 is cause by https://github.com/open-cluster-management-io/ocm/pull/672, 0.31.2 is caused by https://github.com/open-cluster-management-io/ocm/pull/669.

I can try upgrade the other also to 0.31.2 as well.

Reply:
Updated all to 0.31.2.

Reply:
note we also need to upgrade addon-framekwork/sdk-go/API